devdoc/singlylinkedlist2_requirements.md

```diff
@@ -0,0 +1,176 @@
+singlylinkedlist2 requirements
+================
+
+## Overview
+
+SinglyLinkedList is module that provides the functionality of a singly linked list, allowing its user to add, remove and iterate the list elements.
+
+## Exposed API
+
+```c
+typedef struct SLIST_ENTRY_TAG
+{
+    struct SLIST_ENTRY_TAG *next;
+} SLIST_ENTRY, *PSLIST_ENTRY;
+
+typedef bool (*SLIST_MATCH_FUNCTION)(PSLIST_ENTRY listEntry, void* matchContext);
+typedef bool (*SLIST_CONDITION_FUNCTION)(PSLIST_ENTRY listEntry, void* matchContext, bool* continueProcessing);
+typedef int (*SLIST_ACTION_FUNCTION)(PSLIST_ENTRY listEntry, void* action_context, bool* continue_processing);
+
+MOCKABLE_FUNCTION(, void, SList_InitializeListHead, PSLIST_ENTRY, listHead);
```

Comment:
>SList [](http://example.com/codeflow?start=26&length=5)

And naming wise again this would be slist

Reply:
done